src/components/inputs/PrefixedName/useValidatePrefix.ts

```diff
@@ -0,0 +1,157 @@
+import { validateCatalogName } from 'components/inputs/PrefixedName/shared';
+import {
+    PrefixedName_Change,
+    PrefixedName_Errors,
+} from 'components/inputs/PrefixedName/types';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useEntitiesStore_capabilities_adminable } from 'stores/Entities/hooks';
+import { hasLength } from 'utils/misc-utils';
+
+interface Options {
+    allowBlankName?: boolean;
+    allowEndSlash?: boolean;
+    defaultPrefix?: boolean;
+    onChange?: PrefixedName_Change;
+    onNameChange?: PrefixedName_Change;
+    onPrefixChange?: PrefixedName_Change;
+    prefixOnly?: boolean;
+}
+
+function useValidatePrefix({
+    allowBlankName,
+    allowEndSlash,
+    defaultPrefix,
+    onChange,
+    onNameChange,
+    onPrefixChange,
+    prefixOnly,
+}: Options) {
+    const intl = useIntl();
+
+    // Store stuff
+    const adminCapabilities = useEntitiesStore_capabilities_adminable();
+    const objectRoles = Object.keys(adminCapabilities);
+    const singleOption = objectRoles.length === 1;
+
+    // Local State for editing
+    const [errors, setErrors] = useState<string | null>(null);
+    const [name, setName] = useState('');
+    const [nameError, setNameError] = useState<PrefixedName_Errors>(null);
+    const [prefix, setPrefix] = useState(
+        singleOption || defaultPrefix ? objectRoles[0] : ''
+    );
+    const [prefixError, setPrefixError] = useState<PrefixedName_Errors>(null);
+
+    const updateErrors = (prefixValue: string, nameValue: string) => {
+        // Validate both inputs
+        const prefixErrors = validateCatalogName(prefixValue, false, true);
+        const nameErrors = validateCatalogName(
+            nameValue,
+            allowBlankName,
+            allowEndSlash
+        );
+
+        // Array to keep list of errors in by going through each returned
+        //  error and populating with the translated message
+        const updatedErrors: string[] = [];
+        const generateErrorList = (
+            inputName: string,
+            inputErrors: PrefixedName_Errors
+        ) => {
+            inputErrors?.forEach((inputError) => {
+                updatedErrors.push(
+                    intl.formatMessage({
+                        id: `custom.prefixedName.${inputName}.${inputError}`,
+                    })
+                );
+            });
+        };
+
+        // If there are any errors then populate the list
+        if (prefixErrors) {
+            generateErrorList('prefix', prefixErrors);
+        }
+        if (!prefixOnly && nameErrors) {
```

Comment:
Thinking out loud... and I am not sure if I deeply believe this

I wonder if having the settings be more exact like `ignoreName` or something might be a good approach. My main thinking is that is makes it more clear what you're doing and allows for a third option to be added and still work.

However... if we do make this kind of thing `ignoreName` then when adding a third if we wanted to keep this `prefixOnly` we would need to pass `ignoreThird`.